pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -83,10 +84,19 @@ func (lw *logWrapper) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
+// reURL is a regexp to detect if a URL specifies the protocol. We match also strings where the actual protocol is missing
+// (i.e., "://"), in order to catch these as invalid when parsing.
+var reURL = regexp.MustCompile("^[^:]*://")
+
 // NewDataSourceProxy creates a new Datasource proxy
 func NewDataSourceProxy(ds *models.DataSource, plugin *plugins.DataSourcePlugin, ctx *models.ReqContext,
 	proxyPath string, cfg *setting.Cfg) (*DataSourceProxy, error) {
-	targetURL, err := url.Parse(ds.Url)
+	u := ds.Url
+	// Make sure the URL starts with a protocol specifier, so parsing is unambiguous
+	if !reURL.MatchString(u) {
+		u = fmt.Sprintf("http://%s", u)
```

Comment:
Could it cause problems anywhere that the URL's protocol is set to HTTP as a fallback?

Reply:
I _think_ it should be fine as long as the regexp works as expected.

Reply:
I was wondering almost the same; is there any case that the fallback protocol should be `https` instead of `http`?
At least there could be a debug message that the silent implied protocol is `http`?

Reply:
@papagian Would it help in any case to specify https? I am actually hoping that the protocol part of the URL won't be applied anywhere, since chances are it's incorrect. For example, I tested with Postgres, and it doesn't apply the protocol part of the URL (which would fail if it did).

Regarding https, I think it's less safe, since many servers might not support it and if you use http they will typically redirect you to https.

Reply:
I've added debug logging BTW @papagian, please have a look.

Reply:
Honestly I can't tell for sure because I'm not aware of all the possible cases.
I think that the debug message should be enough for now.

I would ideally prefer also a unified `validateUrl` to be used by both the `api` and the `pluginproxy`. What do you think?

Reply:
@papagian I can't share `validateURL` as is, but I can try to extract a common one.

Reply:
not a blocker anyway